File: backend/src/main/java/bag/service/AddressService/AddressService.java
package bag.service.AddressService;

import bag.modal.dto.AddressDto;
import bag.modal.entity.Address;
import bag.modal.request.AddressRequest;

import java.util.List;

public interface AddressService {
    List<AddressDto> getAddresses();

    AddressDto getAddressById(int id);

    // Lấy tất cả địa chỉ của một account
    List<AddressDto> getAddressesByAccountId(int accountId);

    // Lấy địa chỉ mặc định của một account
    AddressDto getDefaultAddress(int accountId);

    // Đặt địa chỉ mặc định
    AddressDto setDefaultAddress(int addressId);

    AddressDto createAddress(AddressRequest request);

    AddressDto updateAddress(AddressRequest request, int id);

    AddressDto deleteAddress(int id);
}

File: frontend/bagshop/src/service/AddressService.js
import axiosClient from "./AxiosConfig";

const AddressService = {
  // Lấy tất cả địa chỉ của một account
  getAddressesByAccountId(accountId) {
    return axiosClient.get(`address/account/${accountId}`);
  },

  // Lấy địa chỉ mặc định của account
  getDefaultAddress(accountId) {
    return axiosClient.get(`address/account/${accountId}/default`);
  },

  // Lấy địa chỉ theo ID
  getAddressById(addressId) {
    return axiosClient.get(`address/${addressId}`);
  },

  // Tạo địa chỉ mới
  createAddress(addressRequest) {
    return axiosClient.post("address/create", addressRequest);
  },

  // Cập nhật địa chỉ
  updateAddress(addressId, addressRequest) {
    return axiosClient.put(`address/${addressId}`, addressRequest);
  },

  // Đặt địa chỉ làm mặc định
  setDefaultAddress(addressId) {
    return axiosClient.put(`address/${addressId}/default`);
  },

  // Xóa địa chỉ
  deleteAddress(addressId) {
    return axiosClient.delete(`address/${addressId}`);
  },
};

export default AddressService;
